### ISSUE TITLE
Retaining on custom dataset

### ISSUE THREAD

**Issue by tejaschandramohan** [zwx8981/DBCNN-PyTorch#26]
Hi. I have a custom dataset consisting of 5248 images. I made necessary changes to include this in data loader. 

In dbcnn.py, I provide the list index as (0,5248). Dbcnn.py runs and takes several hours but does not produce even a single epoch.

When I change the list index as (0,29), then dbcnn.py produces training and val images and the script runs and produces 50 epochs that is set

The images that I use are taken from Android or iphone. Each image have different resolution. Can. You suggest a way to train on my images only? Is there any other way I can increase the speed of training by pre processing my images for your model?

**Comment by zwx8981** [comment]
Did you train our model on CPU or GPU? You'd better debug the program step by step what see in which step the training is stuck.

**Comment by tejaschandramohan** [comment]
It was done on CPU. I did find a correlation on the processing time v/s number of images in the training set. I have now reduced the training images to 2,500 - 2000 training + 500 val. 
Apart from this, conceptually, do you know why DBCNN maybe slow? 
I will check the training script once again. Until then, let this issue be open.

**Comment by zwx8981** [comment]
DBCNN has to branches, its computational complexity is relatively higher. It is highly recommended to train the model on GPU.

**Comment by tejaschandramohan** [comment]
Unfortunately I couldn't get a GPU access until now. Can I train your model on Google Collab? Would you recommend that?

**Comment by zwx8981** [comment]
I think it's a good idea, although I am not familiar with the setup of training models on Google Collab.

**Comment by tejaschandramohan** [comment]
Do you recommend me to keep the same learning rate?

**Comment by tejaschandramohan** [comment]
I tried training your model on the LIVE dataset. Once 50 epochs are done, the training starts again?
Here is the console output
.............................................................................................

Prepare the network and data.
DataParallel(
  (module): DBCNN(
    (features1): Sequential(
      (0): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): ReLU(inplace)
      (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): ReLU(inplace)
      (4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (5): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (6): ReLU(inplace)
      (7): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (8): ReLU(inplace)
      (9): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (10): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (11): ReLU(inplace)
      (12): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (13): ReLU(inplace)
      (14): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (15): ReLU(inplace)
      (16): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (17): Conv2d(256, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (18): ReLU(inplace)
      (19): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (20): ReLU(inplace)
      (21): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (22): ReLU(inplace)
      (23): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (24): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (25): ReLU(inplace)
      (26): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (27): ReLU(inplace)
      (28): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (29): ReLU(inplace)
    )
    (features2): Sequential(
      (0): Conv2d(3, 48, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU(inplace)
      (3): Conv2d(48, 48, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (4): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (5): ReLU(inplace)
      (6): Conv2d(48, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (7): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (8): ReLU(inplace)
      (9): Conv2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (10): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (11): ReLU(inplace)
      (12): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (13): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (14): ReLU(inplace)
      (15): Conv2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (16): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (17): ReLU(inplace)
      (18): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (19): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (20): ReLU(inplace)
      (21): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (22): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (23): ReLU(inplace)
      (24): Conv2d(128, 128, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (25): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (26): ReLU(inplace)
    )
    (fc): Linear(in_features=65536, out_features=1, bias=True)
  )
)
Training.
Epoch	Train loss	Train_SRCC	Test_SRCC	Test_PLCC
*1	1401.993		0.0220		0.4140	0.1699
*2	689.525		0.5538		0.6214	0.5539
*3	634.954		0.8013		0.7906	0.7681
*4	591.026		0.8610		0.8254	0.8106
*5	540.908		0.8604		0.8288	0.8255
*6	494.674		0.8823		0.8343	0.8385
7	481.704		0.8746		0.8320	0.8395
*8	444.264		0.8751		0.8413	0.8453
9	409.228		0.8854		0.8389	0.8469
*10	375.441		0.8902		0.8450	0.8510
11	358.323		0.8850		0.8389	0.8454
12	340.910		0.8831		0.8411	0.8478
*13	327.713		0.8796		0.8518	0.8566
14	299.433		0.8899		0.8420	0.8498
15	293.927		0.8848		0.8416	0.8493
16	266.533		0.8950		0.8431	0.8494
*17	264.365		0.8886		0.8550	0.8620
18	249.650		0.8919		0.8542	0.8579
19	234.825		0.8948		0.8442	0.8520
20	225.925		0.8929		0.8510	0.8585
21	225.488		0.8971		0.8374	0.8464
22	215.468		0.8887		0.8449	0.8511
23	222.953		0.8865		0.8467	0.8542
24	200.109		0.9046		0.8484	0.8574
25	199.094		0.9051		0.8431	0.8538
26	198.172		0.8938		0.8389	0.8486
27	183.668		0.9046		0.8451	0.8550
28	179.099		0.9027		0.8456	0.8587
29	173.531		0.9135		0.8369	0.8488
30	188.271		0.8976		0.8298	0.8431
31	167.976		0.9066		0.8436	0.8493
32	153.070		0.9150		0.8408	0.8500
33	154.369		0.9130		0.8447	0.8579
34	154.272		0.9079		0.8538	0.8615
35	148.291		0.9125		0.8502	0.8601
36	154.517		0.9108		0.8508	0.8593
37	134.769		0.9213		0.8503	0.8600
38	143.584		0.9149		0.8266	0.8355
39	140.327		0.9148		0.8533	0.8619
40	140.768		0.9122		0.8498	0.8627
41	129.561		0.9204		0.8508	0.8592
42	132.605		0.9141		0.8507	0.8634
*43	136.929		0.9125		0.8562	0.8633
44	121.058		0.9240		0.8535	0.8638
45	138.226		0.9139		0.8490	0.8600
*46	129.945		0.9186		0.8571	0.8654
47	118.168		0.9246		0.8543	0.8651
48	122.131		0.9253		0.8529	0.8665
49	122.887		0.9210		0.8545	0.8647
50	114.222		0.9246		0.8508	0.8596
Best at epoch 46, test srcc 0.857112
Prepare the network and data.
DataParallel(
  (module): DBCNN(
    (features1): Sequential(
      (0): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): ReLU(inplace)
      (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): ReLU(inplace)
      (4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (5): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (6): ReLU(inplace)
      (7): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (8): ReLU(inplace)
      (9): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (10): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (11): ReLU(inplace)
      (12): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (13): ReLU(inplace)
      (14): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (15): ReLU(inplace)
      (16): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (17): Conv2d(256, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (18): ReLU(inplace)
      (19): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (20): ReLU(inplace)
      (21): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (22): ReLU(inplace)
      (23): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (24): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (25): ReLU(inplace)
      (26): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (27): ReLU(inplace)
      (28): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (29): ReLU(inplace)
    )
    (features2): Sequential(
      (0): Conv2d(3, 48, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU(inplace)
      (3): Conv2d(48, 48, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (4): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (5): ReLU(inplace)
      (6): Conv2d(48, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (7): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (8): ReLU(inplace)
      (9): Conv2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (10): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (11): ReLU(inplace)
      (12): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (13): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (14): ReLU(inplace)
      (15): Conv2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (16): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (17): ReLU(inplace)
      (18): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (19): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (20): ReLU(inplace)
      (21): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (22): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (23): ReLU(inplace)
      (24): Conv2d(128, 128, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
      (25): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (26): ReLU(inplace)
    )
    (fc): Linear(in_features=65536, out_features=1, bias=True)
  )
)
Training.
Epoch	Train loss	Train_SRCC	Test_SRCC	Test_PLCC
*1	86.879		0.9386		0.8715	0.8837
*2	57.947		0.9592		0.8897	0.8969

**Comment by zwx8981** [comment]
If you only want to train a single time, you can modify the number of the random train/test split to 1

for i in range(0,10): ->  for i in range(0,1):

**Comment by tejaschandramohan** [comment]
I trained the DBCNN model on my custom dataset and the best val srcc I could get is 0.2025, while training on the LIVE Dataset produced 0.8571. 
Here is a screen dump on the custom dataset. Do you recommend me to train for more epochs? I have set the learning rate at 1e-4, batch size is 32. Total number of images is 1500 (train+val)
![image](https://user-images.githubusercontent.com/63153505/142736685-3bf99580-b2d3-4a68-8817-5c3585c6a490.png)
